cylc/flow/broadcast_mgr.py

```diff
@@ -280,8 +280,16 @@ def put_broadcast(
         bad_namespaces = []
 
         with self.lock:
-            for setting in settings:
-                for point_string in point_strings:
+            for setting in settings or []:
```

Comment:
This is 3 levels nested and in theory could be looping a fair bit. Might it be a nicer read and possibly more efficient with:

```
from itertools import product

for setting, point_string, namespace in product(settings or [], point_strings or [], namespaces or []):
    # All below, but with less nesting, and letting C layer do the heavy iterative lifting if necessary.

```

Reply:
I don't *think* product is more efficient (it will still use for loops under the hood, it cannot change the number of iterations), but flattening this logic might be nicer.

I would rather avoid refactors in bugfix PRs through for risk management purposes.